lib/spdy-transport/protocol/http2/parser.js

```diff
@@ -154,9 +154,18 @@ Parser.prototype.onFrameBody = function onFrameBody(header, buffer, callback) {
   else if (header.type === frameType.X_FORWARDED_FOR)
     this.onXForwardedFrame(header, buffer, callback);
   else
-    callback(null, { type: 'unknown: ' + header.type });
+    this.onUnknownFrame(header, buffer, callback);
 };
 
+Parser.prototype.onUnknownFrame = function onUnknownFrame(header, buffer, callback) {
```

Comment:
Ah, looks like we was talking about different things!

So what your patch fixes is:

```
HEADERS
CONTINUATION
UNKNOWN
CONTINUATION
```

Your patch makes this sequence fail. What I would actually like to see it doing is making this fail too:

```
HEADERS
CONTINUATION
WHATEVER NON-HEADERS and NON-CONTINUATION frame
CONTINUATION
```

It doesn't seem to be important that it is not currently checked by h2spec, but if this PR will fix the latter issue - the first issue will be fixed automatically, and we will be more spec compliant!

Does this sound reasonable?

Sorry for long discussion on this, I just want to figure out the best way to do it!